packages/extension/src/view/devtools/components/cookies/cookiesListing/cookieTableContainer/editableComponents/editableCheckBox.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import React, { useCallback, useState, useRef, useEffect } from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import { useCookieStore } from '../../../../../stateProviders/syncCookieStore';
+import classNames from 'classnames';
+import type { InfoType } from '@ps-analysis-tool/design-system';
+import { noop } from '@ps-analysis-tool/common';
+
+interface EditableCheckBoxInputProps {
+  info: InfoType;
+  changedKey: string;
+  cookieKey?: string | null;
+  rowHighlighter?: (value: boolean, cookieKey: string) => void;
+}
+
+const EditableCheckBoxInput = ({
+  info,
+  changedKey,
+  cookieKey,
+  rowHighlighter = noop,
+}: EditableCheckBoxInputProps) => {
+  const [editing, setEditing] = useState(false);
+  const divRef = useRef<HTMLInputElement>(null);
+
+  const [localValue, setLocalValue] = useState<boolean>(info as boolean);
+  const { modifyCookie } = useCookieStore(({ actions }) => ({
+    modifyCookie: actions.modifyCookie,
+  }));
+
+  const handleDoubleClick = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (event.detail === 2) {
+        setEditing(true);
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (editing) {
+      divRef.current?.focus();
+    }
+  }, [editing]);
+
+  const handleClickOutside = useCallback(
+    async (event: MouseEvent) => {
+      if (divRef.current && !divRef.current.contains(event.target as Node)) {
+        setEditing(false);
+        if (localValue !== info && cookieKey) {
+          const result = await modifyCookie(
+            cookieKey,
+            changedKey,
+            localValue,
+            null
+          );
+          if (!result) {
+            rowHighlighter(true, cookieKey);
+            setLocalValue(info as boolean);
+          }
+        }
+      }
+    },
+    [info, changedKey, localValue, modifyCookie, cookieKey, rowHighlighter]
+  );
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
```

Comment:
I guess we can remove this comment, the code explains itself.

Reply:
Added in [8156084](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/278/commits/8156084639e3a4a02224f51f9a795f75e05175ff)